src/actions/actionLock.ts

```diff
@@ -0,0 +1,57 @@
+import { newElementWith } from "../element/mutateElement";
+import { CODES, KEYS } from "../keys";
+import { getSelectedElements } from "../scene";
+import { register } from "./register";
+
+export const actionLock = register({
+  name: "lock",
+  perform: (elements, appState) => {
+    const selectedElements = getSelectedElements(elements, appState, false);
+    const selectedElementIds = new Set(
+      selectedElements.map((element) => element.id),
+    );
+
+    if (selectedElementIds.size === 0) {
+      return false;
+    }
+
+    // It's safe to infer the operation from the first element because selection logic will ensure that selected elements are all of the same locked state
```

Comment:
I need to revisit this comment as it's not exactly correct as seen in the clip below:

https://user-images.githubusercontent.com/9257001/159933610-24efc691-e02c-4fd0-83c5-1108d6c89ad7.mov

There seems to be a strange bug in multi-selection in there too, will look into it.

Reply:
Yes, I'd not depend on element selection order. Let's instead do:

```
const isLockOperation = selectedElements.some(el => !el.locked)
```

Reply:
Also, I'm not sure whether we don't want to split the action into `lock` and `unlock`. Mainly for reasons of a programmatic use (`executeAction()`).

Don't have a strong opinion on this one. But if we're gonna keep it as a single action, then I'd name it `toggleLock` instead.

Reply:
I wasn't sure on that either, I opted for a toggle in the end because I didn't think it made sense to action "unlock" on an already unlocked element.

Reply:
> I wasn't sure on that either, I opted for a toggle in the end because I didn't think it made sense to action "unlock" on an already unlocked element. You'd also need to defer the logic of what to do with mixed lock states to all callsites of the action which doesn't seem ideal.

Yeah, my idea wasn't to support both or push the predicate to callsites. Though since we're using a single shortcut for both actions, we would likely have to put the predicate logic into `keyTest` as well since there's no support for non-contextMenu predicates in actions yet.

Alright, let's keep just a single action, but rename to `toggleLock`?